File: felicia/python/util/command_line_interface/text_style.py
class TextStyle(object):
    RED='\033[91m'
    GREEN='\033[32m'
    BLUE='\033[34m'
    YELLOW='\033[93m'
    BOLD='\033[1m'
    NONE='\033[0m'

    @staticmethod
    def red(stmt):
        return '{}{}{}'.format(TextStyle.RED, stmt, TextStyle.NONE)

    @staticmethod
    def green(stmt):
        return '{}{}{}'.format(TextStyle.GREEN, stmt, TextStyle.NONE)

    @staticmethod
    def blue(stmt):
        return '{}{}{}'.format(TextStyle.BLUE, stmt, TextStyle.NONE)

    @staticmethod
    def yellow(stmt):
        return '{}{}{}'.format(TextStyle.YELLOW, stmt, TextStyle.NONE)

    @staticmethod
    def bold(stmt):
        return '{}{}{}'.format(TextStyle.BOLD, stmt, TextStyle.NONE)

    @staticmethod
    def print_error(stmt):
        print('{} {}'.format(TextStyle.red('Error:'), stmt))

File: felicia/core/util/command_line_interface/text_style.cc
// Copyright (c) 2019 The Felicia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "felicia/core/util/command_line_interface/text_style.h"

namespace felicia {

#define STYLE_METHOD(Style)                              \
  std::string TextStyle::Style(base::StringPiece text) { \
    return ApplyStyle(k##Style, text);                   \
  }

STYLE_METHOD(Red)
STYLE_METHOD(Green)
STYLE_METHOD(Blue)
STYLE_METHOD(Yellow)
STYLE_METHOD(Bold)

#undef STYLE_METHOD

// static
std::string TextStyle::ApplyStyle(const char* style, base::StringPiece text) {
  return base::StrCat({style, text.data(), kNone});
}

}  // namespace felicia